File: 2015/day_01.rb
#!/usr/bin/env ruby

floor = 0

contents = File.read("inputs/day_01.txt")

contents.split("").each_with_index do |c, index|
    if c == "("
        floor = floor + 1
    else
        floor = floor - 1
    end
    if floor == -1
        puts index
    end
end

puts floor

File: 2019/day_01.py
#!/usr/bin/env python

import math

modules = open('inputs/day_01.txt').read().strip().splitlines()


def calculate_fuel(initial):
    fuel = (math.floor(int(initial) / 3) - 2)
    if fuel < 1:
        return 0

    return fuel + calculate_fuel(fuel)


part1 = 0
part2 = 0

for module in modules:
    part1 = part1 + (math.floor(int(module) / 3) - 2)
    part2 = part2 + calculate_fuel(module)

print('Part 1:')
print(part1)

print('Part 2:')
print(part2)




